### ISSUE TITLE
Foreach deconstruction with only discards doesn't parse correctly

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#15451]
```C#

        [Fact]
        public void TwoDiscardsInForeach()
        {
            var source =
@"
class C
{
    static void Main()
    {
        foreach ((_, _) in new[] { (1, ""hello"") }) { System.Console.Write(""2""); }
    }
}
";
            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.DebugExe, references: s_valueTupleRefs);
            comp.VerifyDiagnostics(
                // (6,25): error CS1001: Identifier expected
                //         foreach ((_, _) in new[] { (1, "hello") }) { System.Console.Write("2"); }
                Diagnostic(ErrorCode.ERR_IdentifierExpected, "in").WithLocation(6, 25),
                // (6,25): error CS0230: Type and identifier are both required in a foreach statement
                //         foreach ((_, _) in new[] { (1, "hello") }) { System.Console.Write("2"); }
                Diagnostic(ErrorCode.ERR_BadForeachDecl, "in").WithLocation(6, 25),
                // (6,19): error CS0246: The type or namespace name '_' could not be found (are you missing a using directive or an assembly reference?)
                //         foreach ((_, _) in new[] { (1, "hello") }) { System.Console.Write("2"); }
                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "_").WithArguments("_").WithLocation(6, 19),
                // (6,22): error CS0246: The type or namespace name '_' could not be found (are you missing a using directive or an assembly reference?)
                //         foreach ((_, _) in new[] { (1, "hello") }) { System.Console.Write("2"); }
                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "_").WithArguments("_").WithLocation(6, 22),
                // (6,9): error CS0030: Cannot convert type '(int, string)' to '(_, _)'
                //         foreach ((_, _) in new[] { (1, "hello") }) { System.Console.Write("2"); }
                Diagnostic(ErrorCode.ERR_NoExplicitConv, "foreach").WithArguments("(int, string)", "(_, _)").WithLocation(6, 9)
                );
            // PROTOTYPE(wildcards) Parsing only underscores in foreach doesn't work presently
            // We need at least one declaration expression to confirm it is a d-declaration
            // TODO: This should parse, but report an error if a local called underscore exists.
        }
```

FYI @gafter

**Comment by gafter** [comment]
This should get fixed as a side-effect of my current work (PR published but not yet ready for review)

**Comment by gafter** [comment]
See https://github.com/dotnet/roslyn/blob/f90f62b37e579ca87d096cc71488b0fd966d5d81/src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs#L5015

which shows that this is indeed fixed as a side-effect of #15548

**Comment by gafter** [comment]
Fixed in #15548
